sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -59,8 +62,33 @@ public void enableSessionCapturing(boolean sessionCapturingOverride) {
      *
      * @return the session capturing override
      */
-    public boolean isSessionCapturingOverride() {
-        return this.sessionCapturingOverride;
+    public boolean isSessionCapturingOverrideEnabled() {
+        return this.sessionCapturingOverrideEnabled;
+    }
+
+    /**
+     * Enables transport client sharing allows. The default is false.
+     *
+     * When you have multiple instances of Cosmos Client in the same JVM interacting to multiple Cosmos accounts,
+     * enabling this allows connection sharing in Direct mode if possible between instances of Cosmos Client.
+     * @param transportClientSharingEnabled transport client sharing
+     * @return current cosmosClientBuilder
+     */
+    public CosmosClientBuilder setTransportClientSharingEnabled(boolean transportClientSharingEnabled) {
```

Comment:
Connection sharing assumed with same configuration. Other alternatives of modeling sharing?

Reply:
If connection sharing is enabled, the first config will win. assuming if anyone enables this feature doesn't want different config across multiple clients.

Reply:
Please document the behaviors very explicitly. 
May be easier to say to use it with homogeneous configuration/client-options (first one times might be also ambiguous to them, with concurrent creations).